asreview/webapp/api.py

```diff
@@ -366,6 +367,35 @@ def api_init_model_ready(project_id):  # noqa: F401
     return response
 
 
+@bp.route('/project/<project_id>/model/now_training', methods=["GET"])
+def api_now_training(project_id):
+    """Check if the model is currently being trained.
+
+    But no guarentees on timing.
+    """
+
+    from asreview.webapp.sqlock import get_db
+    lock_fp = get_lock_path(project_id)
+    while True:
+        db = get_db(lock_fp)
+        try:
+            lock_entry = db.execute(
+                'SELECT * FROM locks WHERE name = ?',
+                ("training",)).fetchone()
+            if lock_entry is None:
+                response = jsonify({"status": 0})
+            else:
+                response = jsonify({"status": 1})
+            break
+        except sqlite3.OperationalError as e:
+            print(f"Encountering operational error {e}")
```

Comment:
It would be nice to raise an exception such that we catch it and return an informative API request.

Reply:
I'm not sure what you mean?  It isn't an actual error, just means that the lock file is in use. Right now the API call should timeout when it can't somehow open de database, which is kind of okay-ish?

Reply:
Can't we use https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError when calling the subprocess?

Reply:
I don't think so, since the process asynchronous, but I could be mistaken. Does that exception get raised in the case that the model training fails?